infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsSnapshotReader.java

```diff
@@ -240,17 +233,15 @@ public SnapshotReadMessage read(UUID syncRequestId) {
 
         if (currentStreamHasNext()) {
             msg = read(currentStreamInfo, syncRequestId);
-            if (msg != null) {
```

Comment:
can you explain why this check was removed?

Reply:
here `msg` is always non-null, are we expecting some kind of empty check?